subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/gcc/metadata/GccMetadataProvider.java

```diff
@@ -82,11 +82,32 @@ protected GccMetadata brokenMetadata(String message) {
     @Override
     protected GccMetadata parseCompilerOutput(String output, String error, File gccBinary) {
         Map<String, String> defines = parseDefines(output, gccBinary);
-        VersionNumber versionNumber = determineVersion(defines, gccBinary);
+        VersionNumber scrapedVersionNumber = determineVersion(defines, gccBinary);
         ArchitectureInternal architecture = determineArchitecture(defines);
+        String scrapedVersionNumberString = determineVersionNumberString(error, scrapedVersionNumber, gccBinary);
         ImmutableList<File> systemIncludes = determineSystemIncludes(error);
 
-        return new DefaultGccMetadata(versionNumber, architecture, systemIncludes);
+        return new DefaultGccMetadata(scrapedVersionNumber, scrapedVersionNumberString, architecture, systemIncludes);
+    }
+
+    private String determineVersionNumberString(String error, VersionNumber versionNumber, File gccBinary) {
+        BufferedReader reader = new BufferedReader(new StringReader(error));
```

Comment:
Could we just use `__VERSION__` for this?

Looking at the output of:
```
echo "" | gcc -dM -E - &> gcc
echo "" | clang -dM -E - &> clang
echo "" | gcc-7 -dM -E -v - &> gcc7
```

clang and gcc are the same thing (the output is identical).  `__VERSION__` and `__clang_version__` would work for a "vendor" string.

gcc-7 is from homebrew, but `__VERSION__` is just the version of GCC.  I think this means we'd treat any GCC 7.2.0's as the same thing.  I'd like to see some more examples of the ARM compilers we know about and if we can rely on `__VERSION__` because I think that's going to be good enough for now.

Reply:
On ubuntu we have:
clang:
```
#define __VERSION__ "4.2.1 Compatible Ubuntu Clang 3.6.0 (tags/RELEASE_360/final)"
```
gcc:
```
#define __VERSION__ "4.8.4"
```

That is exactly why I ended up parsing the `-v` output and did not use the macro defines. It seems like we wouldn't be able to distinguish between different GCC vendors - and that is exactly what this change is about.